core/src/main/java/org/jruby/RubyArray.java

```diff
@@ -4639,6 +4640,26 @@ public RubyString pack(ThreadContext context, IRubyObject obj) {
         }
     }
 
+    @JRubyMethod(name = "pack")
+    public RubyString pack(ThreadContext context, IRubyObject obj, IRubyObject maybeOpts) {
+        Ruby runtime = context.runtime;
+        IRubyObject opts = ArgsUtil.getOptionsArg(runtime, maybeOpts);
+        RubyString str = RubyString.newEmptyString(runtime);
+        RubyString iFmt = obj.convertToString();
+        try {
+            if (!opts.isNil()){
+                IRubyObject buffer = ((RubyHash) opts).fastARef(runtime.newSymbol("buffer"));
+                if (buffer != null) {
+                    str = (RubyString) buffer;
```

Comment:
maybe the `RubyString` allocation (above) can be delayed until its sure a buffer is not provided?